### PR TITLE
Feat/scout 178 filters hiding

### PR DESCRIPTION
**Pull request by dejwi**
### Task Description

Dostałem od Jacka sugestię, żeby filtry nie zasłaniały pół ekranu, więc wpakowałem je do accordina i no narazie obczaj sobie bez akceptowania
Dodane na /players

Kwestie czy na każdym formie czy tylko na dłuższych, jak na mobilce itd i czy w ogóle ten format pasuje

### CODE REVIEW

**Comment by dejwi** [comment]
@jakubfiglak na pierwszy rzut oka może wydawać się pogmatwane, w skrócie użyłem nazwy kluczy z obiektu filters, bo prawie zawsze są takie same i we wszystkich combo np jak jest competitionGroupIds to można odrazu wywnioskować docelowy typ obiektu i tym samym wyodrębnić uprganione dane

File: src/components/filter-accordion/getFiltersText.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+export const getFiltersText = (t: TFunction, props: IProps): string[] => {
+  const finalArray: string[] = []
+  const keyTextMapTrans = keyTextMap(t, props)
+  const { filters } = props
+
+  Object.keys(filters).forEach(key => {
+    const value = filters[key]
+    if (!value) return
```
Comment (jakubfiglak) [inline src/components/filter-accordion/getFiltersText.ts:22]: This is why you had this eslint error: `/* eslint-disable consistent-return  */` - one branch of your function returns nothing, other ones return `finalArray`

File: src/components/filter-accordion/getTextFromKey.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    competitionIds: (ids: string[]) => nameRouteArray('COMPETITIONS', ids, data?.competitionsData),
+    countryIds: (ids: string[]) => nameRouteArray('COUNTRIES', ids, data?.countriesData),
+    positionIds: (ids: string[]) => nameRouteArray('POSITIONS', ids, data?.positionsData),
+    teamIds: (ids: string[]) => nameRouteArray('TEAMS', ids, data?.teamsData),
+    bornBefore: (value: number) => `${t('BORN_BEFORE')}: ${value}`,
+    bornAfter: (value: number) => `${t('BORN_AFTER')}: ${value}`,
+    isLiked: () => t('ONLY_LIKED'),
+    footed: (value: string) => `${t('FOOTED')}: ${t(value)}`,
+    name: (value: string) => value,
+  })
```
Comment (jakubfiglak) [inline src/components/filter-accordion/getTextFromKey.tsx:26]: How would you make this reusable?

**Review by jakubfiglak — CHANGES_REQUESTED** [review]
I don't think we can accomplish want we want regarding displaying applied filters with this approach. What I'd suggest to focus on within this PR is to create UI for hiding filter forms in the accordion. We can tackle displaying applied filters when we're done with this refactor https://github.com/scoutmaker/scoutmaker-web-v2/pull/54. 

For now, I'm not sure if I like dark background on the `AccordionSummary`, please try and use the default one. Also, please use `expandIcon` on this component.

File: src/components/filter-accordion/getTextFromKey.tsx (hunk 2 above)
Comment (dejwi) [reply src/components/filter-accordion/getTextFromKey.tsx:26]: problem is IT'S ALREADY REUSABLE - I'll explain more in a second

**Comment by dejwi** [comment]
I think there's been a misunderstanding because it's already fully REUSABLE.
I've made 2 commits that adds FULL functionality to `/user-subscriptions`.
First commit exits only because for now I made types/keyTextMap applicable fully only to `/players`, so when everything is added second commit is what's needed to reuse my accordion.
PLEASE CHECK🙏🙏

File: src/components/filter-accordion/getFiltersText.ts (hunk 1 above)
Comment (dejwi) [reply src/components/filter-accordion/getFiltersText.ts:22]: `return finalArray` is outside this function and empty returns are just to skip to the next iteration(as basicly all returns there)

I broke the DRY rule here but it's for tomorrow😅

Comment (dejwi) [reply src/components/filter-accordion/getFiltersText.ts:22]: I understand that but why value of return matters on `.forEach` 😅

I broke the DRY rule here but it's for tomorrow

File: src/components/filter-accordion/getTextFromKey.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    bornAfter: (value: number) => `${t('BORN_AFTER')}: ${value}`,
+    isLiked: () => t('ONLY_LIKED'),
+    footed: (value: string) => `${t('FOOTED')}: ${t(value)}`,
+    name: (value: string) => value,
+    userId: (id: string) => {
+      if (!data?.usersData) return 'NO DATA'
+      const user = data.usersData.find(u => u.id === id)
+      return `${t('USER')}: ${user?.firstName} ${user?.lastName}`
+    },
+  })
```
Comment (jakubfiglak) [inline src/components/filter-accordion/getTextFromKey.tsx:31]: Sorry but I cannot agree that this is reusable:) What if we had 200 different filter types in our application? Or a 1000? I mean, you're doing a great job trying to figure this out but I think we should change the approach here, I will explain what architectural issue I see in review summary.

**Review by jakubfiglak — CHANGES_REQUESTED** [review]
Once again - great job trying to figure this out but I don't think this is a right solution to our problem. We have a general issue with how we handle filter data, let's take teams list data as an example. Currently the flow looks like this:

- we fetch data from `/teams/list` endpoint - response is of a type `TeamBasicDataDto[]`,
- we pass this data to `TeamsCombo` component, we extract the id, we keep this as the input value, and do some crazy shit to get a display label:

```
      getOptionLabel={(option: string) => {
        if (option === '') {
          return ''
        }
        const team = data.find(c => c.id === option)
        if (team) {
          return team.name
        }
        return t('NONE')
      }}
```
- our filter state is of a type `string[]`
- now we want to display filters in a user friendly way (user doesn't care about ids, for them it's just a random jibber jabber)
- we need to do crazy shit AGAIN to basically GET BACK from `string[]` to `TeamBasicDataDto[]`

We can avoid doing all this work by simply keeping `TeamBasicDataDto` in our filter state. We could then display it if we want, map to ids and send to backend. This is actually much more flexible - we can also keep more data in state and for example conditionally render some filter inputs (like it's been requested in this ticket - https://playmakerpro.atlassian.net/browse/SCOUT-186). It would not be possible with our current approach. I know it requires big refactor but I think it's worth it. Let me know what you think.

**Comment by jakubfiglak** [comment]
Please do not abandon the work you've done here. The ticket (https://playmakerpro.atlassian.net/browse/SCOUT-178) is about providing option to hide filters, so let's focus on that - put the form inside the accordion and that's it. We will worry about displaying applied filters later.

**Comment by dejwi** [comment]
By default you mean this or white? and do we keep `changed` chip for now or delete it
image

**Review by jakubfiglak — COMMENTED** [review]
I'd get rid of the chip for now. And I'd apply all these changes to every module.

BTW, what if we do another thing - we can make the filter form full screen width for desktop and place inputs side by side? What do you think? I believe Jacek's already suggested such a change. Can you try it?

**Comment by jakubfiglak** [comment]
> Side by side you mean like 2 columns ?

Maybe even more? I'm not sure what's going to look and feel nice, I encourage you to experiment. You can go to some well known websites and look at their ui implementation, here's Zalando for example:

Screenshot 2022-09-30 at 14 41 59

Screenshot 2022-09-30 at 14 42 33

**Comment by dejwi** [comment]
@jakubfiglak what do you think of this?? `/players` + it's pretty responsive
flex breakpoint in form container is for keeping columns at max '3'  as above that it looks ugly

Maybe we could change this footed-select to combo ? This misaligned text bugs me a lot

note: had to change mui grid to box because of this:
image

**Comment by dejwi** [comment]
I've made the mistake of editing form container directly again I'll separate it but `/players` POC works😅

**Comment by dejwi** [comment]
Yes I've tried to fix it but currently failed to
What's the problem `flexBasis` ? It scales and basically regulates the number of columns as it's not grid to achieve this cool filled effect
^made some fixes

File: src/components/filter-accordion/filter-accordion.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+const StyledAccordion = styled(Accordion)(({ theme }) => ({
+  margin: theme.spacing(2, 0),
+  background: 'none',
+}))
+
+const StyledAccordionDetails = styled(AccordionDetails)(({ theme }) => ({
+  paddingTop: theme.spacing(2),
+}))
+
+const StyledAccordinText = styled(Typography)({
```
Comment (jakubfiglak) [inline src/components/filter-accordion/filter-accordion.tsx:35]: There's still a typo :D

File: src/components/filter-accordion/filter-accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  fontWeight: 'bold',
+  textAlign: 'center',
```
Comment (jakubfiglak) [inline src/components/filter-accordion/filter-accordion.tsx:37]: I'd left align it personally.

**Review by jakubfiglak — COMMENTED** [review]
Looks good to me. I think we can apply this to all filter forms throughout the app. 

One thing though - I think you can try and make this accordion look a little bit nicer. I know I told you to change the background colour to white but it was not a good idea - it should contrast from the background. And I think it'd be good if it had `border-radius` exactly like the table below. Just try and make it look nice please :D

**Comment by dejwi** [comment]
I think that center text looks better, left align isn't included in new commits
image

**Review by jakubfiglak — APPROVED** [review]

**Pull request merged**